validphys2/src/validphys/photon/compute.py

```diff
@@ -0,0 +1,312 @@
+        photon_100GeV /= self.xgrid
+        # TODO : the different x points could be even computed in parallel
+
+        # Load eko and reshape it
+        with EKO.read(self.path_to_eko_photon) as eko:
+            # If we make sure that the grid of the precomputed EKO is the same of
+            # self.xgrid then we don't need to reshape
+            # TODO : move the reshape inside vp-setupfit
+            # xgrid_reshape(eko, targetgrid = XGrid(self.xgrid), inputgrid = XGrid(self.xgrid))
+
+            # construct PDFs
+            pdfs = np.zeros((len(eko.rotations.inputpids), len(self.xgrid)))
+            for j, pid in enumerate(eko.rotations.inputpids):
+                if pid == 22:
+                    pdfs[j] = photon_100GeV
+                    ph_id = j
+                if pid not in self.qcd_pdfs.flavors:
+                    continue
+                pdfs[j] = np.array(
+                    [self.qcd_pdfs.xfxQ(x, self.q_in, id, pid) / x for x in self.xgrid]
+                )
+
+            # Apply EKO to PDFs
+            q2 = eko.mu2grid[0]
+            with eko.operator(q2) as elem:
+                pdf_final = np.einsum("ajbk,bk", elem.operator, pdfs)
+                # error_final = np.einsum("ajbk,bk", elem.error, pdfs)
+
+        photon_Q0 = pdf_final[ph_id]
+
+        # we want x * gamma(x)
+        return self.xgrid * photon_Q0
+
+    def produce_interpolators(self):
+        """Produce the interpolation functions to be called in compute."""
+        self.photons_array = [self.compute_photon_array(id) for id in self.replicas_id]
+        self.interpolator = [
+            interp1d(self.xgrid, photon_array, fill_value=0.0, kind="cubic")
+            for photon_array in self.photons_array
+        ]
+
+    def compute(self, xgrid):
+        """
+        Compute the photon interpolating the values of self.photon_array.
+
+        Parameters
+        ----------
+        xgrid : nd.array
+            array of x values with shape (1,xgrid,1)
+
+        Returns
+        -------
+        photon values : nd.array
+            array of photon values with shape (1,xgrid,1)
+        """
+        return [
+            self.interpolator[id](xgrid[0, :, 0])[np.newaxis, :, np.newaxis]
+            for id in range(len(self.replicas_id))
+        ]
+
+    def integrate(self):
+        """Compute the integral of the photon on the x range."""
+        return [
+            trapezoid(self.photons_array[id], self.xgrid) for id in range(len(self.replicas_id))
+        ]
+
+    def generate_error_matrix(self):
+        """generate error matrix to be used for the additional errors."""
+        if not self.fiatlux_runcard["additional_errors"]:
+            return None
+        extra_set = LHAPDFSet("LUXqed17_plus_PDF4LHC15_nnlo_100", "replicas")
+        qs = [self.q_in] * len(self.xgrid)
+        res_central = np.array(extra_set.central_member.xfxQ(22, self.xgrid, qs))
+        res = []
+        for idx_member in range(101, 107 + 1):
```

Comment:
I guess replicas 101-107 are special error members for `LUXqed17_plus_PDF4LHC15_nnlo_100`, so add a comment explaining this (also maybe move it module level as well). 

It looks like this function creates some sort of theory matrix, but exactly how/why/what it is, is not clear. Could you add a reference?